sphinx_autobuild/build.py

```diff
@@ -29,12 +30,23 @@ def __init__(self, watcher, sphinx_args, *, host, port, pre_build_commands):
         self.sphinx_args = sphinx_args
         self.pre_build_commands = pre_build_commands
         self.uri = f"http://{host}:{port}"
+        self.files = []
+        self.ioloop = ioloop.IOLoop.instance()
 
     def __call__(self):
         """Generate the documentation using ``sphinx``."""
 
-        if self.watcher.filepath:
-            show(context=f"Detected change: {self.watcher.filepath}")
+        self.files.append(self.watcher.filepath)
+        if len(self.files) == 1:
+            self.ioloop.add_callback(self.callback)
+
+    def callback(self):
+        if self.files:
+            for file in self.files:
+                if file:
+                    show(context=f"Detected change: {file}")
+
+            self.files = []
```

Comment:
> This seems unrelated?

It could be split into a separate PR if that'd be helpful.  The commit explains it here: https://github.com/sphinx-doc/sphinx-autobuild/pull/151/commits/0c2914fd79d1381debee6847f774cf04d9c863fc

Mainly I've encountered situations where multiple files got touched causing the `livereload` layer to call us back multiple times in a row each time causing a build.  This is really annoying when building a longer build target, such as when using `-M latexpdf`.  It's not as annoying with the `-b html` target since Sphinx doesn't rebuild the entire doctree for that target (but only the changed files).  The `-b latex` (and thus also `-M latexpdf`) causes the entire doctree to be rebuilt and is terribly slow if that occurs.

When `pyinotify` is installed, the callbacks from tornado occur back to back in a row and if we install a callback handler via the tornado ioloop it'll get invoked *after* all of the callbacks that were generated at the same time.  This doesn't work for the file polling method of `livereload` (e.g. when `pyinotify` is not installed) but it doesn't introduce any adverse behaviors by adding this callback hook at the top level.